debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```
@@ -0,0 +1,126 @@
+        else {
+            return "application/json";
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        HttpClient client = HttpClient.newHttpClient();
+
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+
+            try {
+                HttpRequest.Builder requestBuilder = HttpRequest.newBuilder(new URI(sinkUrl))
+                        .timeout(timeoutDuration);
+
+                String value = "";
+                if (record.value() != null) {
+                    value = (String) record.value();
+
+                    // Since the ChangeEvent has already been serialized at this point, we need to inspect it to see if this is
+                    // a cloudevent, and set the correct content-type header.
+                    requestBuilder.setHeader("content-type", getContentType(value.contains("\"specversion\":\"1.0\"")));
+
+                    HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(value)).build();
+                    client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                            .thenAccept(r -> {
+
+                                if ((r.statusCode() == HTTP_OK) || (r.statusCode() == HTTP_NO_CONTENT)) {
+                                    try {
+                                        committer.markProcessed(record);
+                                    }
+                                    catch (InterruptedException e) {
+                                        LOGGER.info("exception while committing work: " + e.toString());
+                                    }
+                                }
+                                else {
+                                    LOGGER.info("failed to publish event: " + r.body());
```

Should we rather raise an exception in this case, also including the status code?

---

Didn't think it'd raise to the level of throwing an exception given the service could be temporarily unavailable, but still allow for a chance to retry sending the event.